src/main.rs

```diff
@@ -68,29 +66,46 @@ fn main() {
 
 /// A Shop is a collection of 3 cards.
 struct Shop {
-    // TOOD: Add the field to this struct
+    cards: [Card; 3],
 }
 
 impl Shop {
     /// Get the price of the most expensive card in the shop
     fn most_expensive(&self) -> u32 {
-        todo!()
+        let mut h: u32 = 0;
+        for card in self.cards.iter() {
+            if card.price > h {
+                h = card.price;
+            }
+        }
+        h
     }
 
     /// Get the total damage of all cards in the shop
     fn total_damage(&self) -> u32 {
-        todo!()
+        let mut damage: u32 = 0;
+        for card in self.cards.iter() { 
+            damage += card.damage;
+        }
+        damage
     }
 
     /// Get the total health of all cards in the shop
     fn total_health(&self) -> u32 {
-        todo!()
+        let mut health: u32 = 0;
+        for card in self.cards.iter() { 
+            health += card.health;
+        }
+        health
     }
+    
 }
 
 /// A Card is a card stores a price, health, and damage.
-struct Card {
-    // TODO: Add fields to this struct
+struct Card { //prob overkill but idk what kinda numbers we wroking with here practically
+    price: u32,
+    damage: u32, 
+    health: u32,
```

Comment:
This is a good thought, but when keeping this too much in mind, if often leads to over-engineering before we know how the problem is actually solved :) it would be an easy bound to fix if it did actually cause a performance issue :100: